docs/design/proposal/dependency-upgrades/aws-sdk-upgrades.md

```diff
@@ -0,0 +1,152 @@
+release
+
+###### Setup
+There will be a new periodic prowjob which will run once every 24 hours.
+This job will checkout all the service controller repos, along with latest
+ACK runtime release and corresponding code-generator release.
+
+> NOTE: We will not checkout `main` branch of ACK core libraries because this
+job should only validate the changes caused by aws-sdk-go upgrades for
+the latest controller release
+
+###### Controller Build
+* In the periodic job mentioned above, the ENTRYPOINT script will first
+validate whether there is a new minor version upgrade release for aws-sdk-go
+> NOTE: ACK runtime does not need to be updated for every aws-sdk-go
+> patch release since only *SDK Feature* updates and IAM service changes
+> impact ACK runtime functionality
+
+* If there is a new aws-sdk-go minor-version-upgrade release present,
+update the ACK runtime locally and validate that it compiles successfully
+* Update the ACK code-generator `go.mod` file to replace ACK runtime
+dependency with local ACK runtime repo and validate that it compiles
+successfully. (The ACK runtime update will bring new aws-sdk-go dependency)
+* Perform similar go.mod update for all service controllers and execute
+`make build-controller` target for each service controller.
+* Perform `make test` target for each controller to validate that newly
+generated controller compiles successfully.
+
+###### Reporting Results
+* This step can be started as minimal reporting using `STDOUT` at the end of
+prowjob. As the number of service controllers increase, the result can be
+condensed into a separate summary file and individual results can be exported
+into separate files per controller and stored in a S3 bucket.
+* For every aws-sdk-go release upgrade, There will be auto created GitHub issue
+to track the actual release and these artifacts will be linked in that issue.
+* In the future, if needed, we can have a static webpage that can help service teams
+and external contributors browse these artifacts before the upgrade.
+
+### Release
+With the automation is place for **auto-generating** service controllers, this step can
+be started with some manual effort and completely automated with future enhancements.
+
+Currently, after reviewing the artifacts of "Verification" step from the GitHub
+issue, ACK core team member can just update the aws-sdk-go version manually in
+the ACK runtime repo and follow the existing release process, which will auto-generate
+new service controllers.
+
+In future enhancements, the Verification step will also generate the Pull Request with
+new aws-sdk-go update for ACK runtime. Creating new ACK runtime patch release, updating
+code-generator and creating new code-generator release will also be automated when runtime
+PR gets merged.
+
+
+### Monitoring
+* Since this is one of the important jobs, when this job fails ACK core team should be
+updated by some mechanism.
```

Comment:
What do you mean by "this is one of the most important jobs"? Why is checking for minor updates of the SDK so important?

Reply:
> "this is one of the ~most~ important jobs"

This seemed important to me because this is something we never pay attention to and can easily be missed if the prow job keeps failing. 
To your point, there is nothing very important with minor-version upgrade, it's a nice to have feature.